spring-boot-project/spring-boot-actuator-autoconfigure/src/main/java/org/springframework/boot/actuate/autoconfigure/metrics/jdbc/DataSourcePoolMetricsAutoConfiguration.java

```diff
@@ -68,7 +68,7 @@
 		@Autowired
 		void bindDataSourcesToRegistry(Map<String, DataSource> dataSources, MeterRegistry registry,
 				ObjectProvider<DataSourcePoolMetadataProvider> metadataProviders) {
-			List<DataSourcePoolMetadataProvider> metadataProvidersList = metadataProviders.stream()
+			List<DataSourcePoolMetadataProvider> metadataProvidersList = metadataProviders.orderedStream()
```

Comment:
Does ordering matter here? The providers are used in a first wins manner. For the ordering to be important there would have to be overlap between providers.